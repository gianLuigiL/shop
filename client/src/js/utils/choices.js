export const choices = [
    {display_value: "Red", value: "red", checked: false},
    {display_value: "Green", value: "green", checked: false}
];