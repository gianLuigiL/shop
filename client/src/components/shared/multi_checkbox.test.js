import React from 'react'
import { shallow, mount } from "enzyme";
import MultiCheckbox from "./multi_checkbox";
import { choices } from '../../js/utils/choices';


describe("MultiCheckbox", () => {
    it("Should render correctly without props", () => {
        const component = shallow(<MultiCheckbox/>);
        expect(component).toMatchSnapshot();
    });

    it("Should render correctly with props", () => {
        const component = shallow(<MultiCheckbox choices={choices}/>);
        expect(component).toMatchSnapshot();
    });

    it("Should render an input for every choice", () => {
        const component = mount(<MultiCheckbox choices={choices}/>);
        expect(component.find("input").length).toBe(choices.length);
        component.unmount();
    });
    it("Should call the handler", () => {
        const change_handler = jest.fn();
        const component = mount(<MultiCheckbox choices={choices} change_handler={change_handler}/>);
        component.find(`input#${choices[0].value}`).simulate("change");
        expect(change_handler).toHaveBeenCalled();
        component.unmount();
    });
})