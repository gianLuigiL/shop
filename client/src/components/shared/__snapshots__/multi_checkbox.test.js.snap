// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiCheckbox Should render correctly with props 1`] = `
<span
  className="multi_checkbox_container horizontal"
>
  <span
    className="multi_checkbox"
    key="red"
  >
    <label
      className="d-flex flex-wrap align-items-center justify-content-start"
      htmlFor="red"
    >
      <input
        defaultChecked=""
        id="red"
        name="red"
        onChange={[Function]}
        tabIndex="0"
        type="checkbox"
        value="red"
      />
      <span
        className="custom_checkbox"
      />
      <span
        className="text"
      >
        Red
      </span>
    </label>
  </span>
  <span
    className="multi_checkbox"
    key="green"
  >
    <label
      className="d-flex flex-wrap align-items-center justify-content-start"
      htmlFor="green"
    >
      <input
        defaultChecked=""
        id="green"
        name="green"
        onChange={[Function]}
        tabIndex="0"
        type="checkbox"
        value="green"
      />
      <span
        className="custom_checkbox"
      />
      <span
        className="text"
      >
        Green
      </span>
    </label>
  </span>
</span>
`;

exports[`MultiCheckbox Should render correctly without props 1`] = `
<span
  className="multi_checkbox_container horizontal"
/>
`;
